Data source "teams" produces stacktrace if "params" are set

Recreated issue due to response to issue [#101](https://github.com/salesforce/terraform-provider-anypoint/issues/101#issuecomment-3327387683) of project [salesforce/terraform-provider-anypoint](https://github.com/salesforce/terraform-provider-anypoint).

The `anypoint_teams` data source produces the below console output, if `params` like `offset` and `limit` are set:

```text
$ tofu plan -var client_id=<connected_app_client_id> -var client_secret=<connected_app_client_secret> -var organization_id=<organization_id>
data.anypoint_teams.teams: Reading...

Planning failed. OpenTofu encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│
│   with data.anypoint_teams.teams,
│   on main.tf line 24, in data "anypoint_teams" "teams":
│   24: data "anypoint_teams" "teams" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-anypoint_v1.8.0 plugin:

panic: interface conversion: interface {} is []interface {}, not []string

goroutine 81 [running]:
github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint.parseTeamSearchOpts({{0x135d490, 0xc0000cfe00}, 0xc000583328, {0xc0002be120, 0x24}, 0x0, 0x0, 0xc0007c4a20, 0x0, 0x0, ...}, ...)
        github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint/data_source_teams.go:234 +0x974
github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint.dataSourceTeamsRead({0x135d538, 0xc0002d6230}, 0xc0001c7480, {0x11baea0?, 0xc000168b40?})
        github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint/data_source_teams.go:165 +0x2a5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0000007e0, {0x135d490, 0xc00071f260}, 0xc0001c7480, {0x11baea0, 0xc000168b40})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:724 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0000007e0, {0x135d490, 0xc00071f260}, 0xc0001c7300, {0x11baea0, 0xc000168b40})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:943 +0x127
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000519ab8, {0x135d490?, 0xc00071f1a0?}, 0xc000990880)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1179 +0x346
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc000484780, {0x135d490?, 0xc00071e9c0?}, 0xc000986690)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:658 +0x3c4
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x117a000, 0xc000484780}, {0x135d490, 0xc00071e9c0}, 0xc0002d61c0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005303c0, {0x1362200, 0xc00012e000}, 0xc000666d80, 0xc000437c20, 0x1be1e70, 0x0)
        google.golang.org/grpc@v1.48.0/server.go:1295 +0xade
google.golang.org/grpc.(*Server).handleStream(0xc0005303c0, {0x1362200, 0xc00012e000}, 0xc000666d80, 0x0)
        google.golang.org/grpc@v1.48.0/server.go:1636 +0x9cc
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.48.0/server.go:932 +0x87
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 43
        google.golang.org/grpc@v1.48.0/server.go:930 +0x226

Error: The terraform-provider-anypoint_v1.8.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

It seems that some default values have not been properly defined, because the plugin stops at statement 234, although `ancestor_team_id` has not be specified:

```text
	for k, v := range opts.(map[string]interface{}) {
		if k == "ancestor_team_id" {
			req = req.AncestorTeamId(v.([]string)) # This is line 234.
			continue
		}
```

ToFu version is **v1.9.0**.
Anypoint provider version is **1.8.0**.
Attachment: [main_tf.txt](https://github.com/user-attachments/files/22513442/main_tf.txt)

--- a/anypoint/data_source_team.go
+++ b/anypoint/data_source_team.go
@@ -80,7 +80,7 @@ func dataSourceTeamRead(ctx context.Context, d *schema.ResourceData, m any) diag
 	teamid := d.Get("id").(string)
 	authctx := getTeamAuthCtx(ctx, &pco)
 	//request roles
-	res, httpr, err := pco.teamclient.DefaultApi.OrganizationsOrgIdTeamsTeamIdGet(authctx, orgid, teamid).Execute()
+	res, httpr, err := pco.teamclient.DefaultApi.GetTeam(authctx, orgid, teamid).Execute()
 	if err != nil {
 		var details string
 		if httpr != nil && httpr.StatusCode >= 400 {
@@ -99,7 +99,7 @@ func dataSourceTeamRead(ctx context.Context, d *schema.ResourceData, m any) diag
 	}
 	defer httpr.Body.Close()
 	//process data
-	team := flattenTeamData(&res)
+	team := flattenTeamData(res)
 	//save in data source schema
 	if err := setTeamAttributesToResourceData(d, team); err != nil {
 		diags = append(diags, diag.Diagnostic{
@@ -133,7 +133,7 @@ func flattenTeamData(team *team.Team) map[string]any {
 		item["team_type"] = *val
 	}
 	if val, ok := team.GetAncestorTeamIdsOk(); ok {
-		item["ancestor_team_ids"] = *val
+		item["ancestor_team_ids"] = val
 	}
 	if val, ok := team.GetCreatedAtOk(); ok {
 		item["created_at"] = *val

--- a/anypoint/data_source_teams.go
+++ b/anypoint/data_source_teams.go
@@ -231,11 +231,21 @@ func parseTeamSearchOpts(req team.DefaultApiApiOrganizationsOrgIdTeamsGetRequest
 
 	for k, v := range opts.(map[string]any) {
 		if k == "ancestor_team_id" {
-			req = req.AncestorTeamId(v.([]string))
+			ati := v.([]any)
+			ati2 := make([]string, len(ati))
+			for i, v := range ati {
+				ati2[i] = v.(string)
+			}
+			req = req.AncestorTeamId(ati2)
 			continue
 		}
 		if k == "parent_team_id" {
-			req = req.ParentTeamId(v.([]string))
+			pati := v.([]any)
+			pati2 := make([]string, len(pati))
+			for i, v := range pati {
+				pati2[i] = v.(string)
+			}
+			req = req.ParentTeamId(pati2)
 			continue
 		}
 		if k == "team_id" {

--- a/anypoint/data_source_teams.go
+++ b/anypoint/data_source_teams.go
@@ -54,7 +54,7 @@ func dataSourceTeams() *schema.Resource {
 						"team_type": {
 							Type:        schema.TypeString,
 							Optional:    true,
-							Description: "return only teams that are of this type",
+							Description: "return only teams that are of the given type. Enum values are: internal, shared, private, external.",
 						},
 						"search": {
 							Type:        schema.TypeString,
@@ -76,7 +76,7 @@ func dataSourceTeams() *schema.Resource {
 						"sort": {
 							Type:        schema.TypeString,
 							Optional:    true,
-							Description: "The field to sort on.",
+							Description: "The field to sort on. Default is team_name.",
 						},
 						"ascending": {
 							Type:        schema.TypeBool,
@@ -121,7 +121,7 @@ func dataSourceTeams() *schema.Resource {
 						},
 						"ancestor_team_ids": {
 							Type:     schema.TypeList,
-							Optional: true,
+							Computed: true,
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
 							},
@@ -161,7 +161,7 @@ func dataSourceTeamsRead(ctx context.Context, d *schema.ResourceData, m any) dia
 	orgid := d.Get("org_id").(string)
 	authctx := getTeamAuthCtx(ctx, &pco)
 	//prepare request
-	req := pco.teamclient.DefaultApi.OrganizationsOrgIdTeamsGet(authctx, orgid)
+	req := pco.teamclient.DefaultApi.GetTeams(authctx, orgid)
 	req, errDiags := parseTeamSearchOpts(req, searchOpts)
 	if errDiags.HasError() {
 		diags = append(diags, errDiags...)
@@ -221,7 +221,7 @@ func dataSourceTeamsRead(ctx context.Context, d *schema.ResourceData, m any) dia
 	return diags
 }
 
-func parseTeamSearchOpts(req team.DefaultApiApiOrganizationsOrgIdTeamsGetRequest, params *schema.Set) (team.DefaultApiApiOrganizationsOrgIdTeamsGetRequest, diag.Diagnostics) {
+func parseTeamSearchOpts(req team.DefaultApiGetTeamsRequest, params *schema.Set) (team.DefaultApiGetTeamsRequest, diag.Diagnostics) {
 	var diags diag.Diagnostics
 	if params.Len() == 0 {
 		return req, diags
@@ -230,7 +230,7 @@ func parseTeamSearchOpts(req team.DefaultApiApiOrganizationsOrgIdTeamsGetRequest
 	opts := params.List()[0]
 
 	for k, v := range opts.(map[string]any) {
-		if k == "ancestor_team_id" {
+		if k == "ancestor_team_id" && len(v.([]any)) > 0 {
 			ati := v.([]any)
 			ati2 := make([]string, len(ati))
 			for i, v := range ati {
@@ -239,7 +239,7 @@ func parseTeamSearchOpts(req team.DefaultApiApiOrganizationsOrgIdTeamsGetRequest
 			req = req.AncestorTeamId(ati2)
 			continue
 		}
-		if k == "parent_team_id" {
+		if k == "parent_team_id" && len(v.([]any)) > 0 {
 			pati := v.([]any)
 			pati2 := make([]string, len(pati))
 			for i, v := range pati {
@@ -248,15 +248,15 @@ func parseTeamSearchOpts(req team.DefaultApiApiOrganizationsOrgIdTeamsGetRequest
 			req = req.ParentTeamId(pati2)
 			continue
 		}
-		if k == "team_id" {
+		if k == "team_id" && v.(string) != "" {
 			req = req.TeamId(v.(string))
 			continue
 		}
-		if k == "team_type" {
+		if k == "team_type" && v.(string) != "" {
 			req = req.TeamType(v.(string))
 			continue
 		}
-		if k == "search" {
+		if k == "search" && v.(string) != "" {
 			req = req.Search(v.(string))
 			continue
 		}
@@ -268,7 +268,7 @@ func parseTeamSearchOpts(req team.DefaultApiApiOrganizationsOrgIdTeamsGetRequest
 			req = req.Limit(int32(v.(int)))
 			continue
 		}
-		if k == "sort" {
+		if k == "sort" && v.(string) != "" {
 			req = req.Sort(v.(string))
 			continue
 		}

--- a/anypoint/resource_team.go
+++ b/anypoint/resource_team.go
@@ -99,7 +99,7 @@ func resourceTeamCreate(ctx context.Context, d *schema.ResourceData, m any) diag
 	authctx := getTeamAuthCtx(ctx, &pco)
 	body := newTeamPostBody(d)
 	//request user creation
-	res, httpr, err := pco.teamclient.DefaultApi.OrganizationsOrgIdTeamsPost(authctx, orgid).TeamPostBody(*body).Execute()
+	res, httpr, err := pco.teamclient.DefaultApi.CreateTeam(authctx, orgid).TeamPostBody(*body).Execute()
 	if err != nil {
 		var details string
 		if httpr != nil && httpr.StatusCode >= 400 {
@@ -134,7 +134,7 @@ func resourceTeamRead(ctx context.Context, d *schema.ResourceData, m any) diag.D
 	}
 	authctx := getTeamAuthCtx(ctx, &pco)
 	//request roles
-	res, httpr, err := pco.teamclient.DefaultApi.OrganizationsOrgIdTeamsTeamIdGet(authctx, orgid, teamid).Execute()
+	res, httpr, err := pco.teamclient.DefaultApi.GetTeam(authctx, orgid, teamid).Execute()
 	if err != nil {
 		var details string
 		if httpr != nil && httpr.StatusCode >= 400 {
@@ -153,7 +153,7 @@ func resourceTeamRead(ctx context.Context, d *schema.ResourceData, m any) diag.D
 	}
 	defer httpr.Body.Close()
 	//process data
-	team := flattenTeamData(&res)
+	team := flattenTeamData(res)
 	//save in data source schema
 	if err := setTeamAttributesToResourceData(d, team); err != nil {
 		diags = append(diags, diag.Diagnostic{
@@ -180,7 +180,7 @@ func resourceTeamUpdate(ctx context.Context, d *schema.ResourceData, m any) diag
 	if d.HasChanges(getTeamPatchWatchAttributes()...) {
 		body := newTeamPatchBody(d)
 		//request user creation
-		_, httpr, err := pco.teamclient.DefaultApi.OrganizationsOrgIdTeamsTeamIdPatch(authctx, orgid, teamid).TeamPatchBody(*body).Execute()
+		_, httpr, err := pco.teamclient.DefaultApi.UpdateTeam(authctx, orgid, teamid).TeamPatchBody(*body).Execute()
 		if err != nil {
 			var details string
 			if httpr != nil && httpr.StatusCode >= 400 {
@@ -204,7 +204,7 @@ func resourceTeamUpdate(ctx context.Context, d *schema.ResourceData, m any) diag
 	if d.HasChanges(getTeamPutWatchAttributes()...) {
 		body := newTeamPutBody(d)
 		//request user creation
-		_, httpr, err := pco.teamclient.DefaultApi.OrganizationsOrgIdTeamsTeamIdParentPut(authctx, orgid, teamid).TeamPutBody(*body).Execute()
+		_, httpr, err := pco.teamclient.DefaultApi.MoveTeam(authctx, orgid, teamid).TeamPutBody(*body).Execute()
 		if err != nil {
 			var details string
 			if httpr != nil && httpr.StatusCode >= 400 {
@@ -235,7 +235,7 @@ func resourceTeamDelete(ctx context.Context, d *schema.ResourceData, m any) diag
 	orgid := d.Get("org_id").(string)
 	authctx := getTeamAuthCtx(ctx, &pco)
 	//perform request
-	httpr, err := pco.teamclient.DefaultApi.OrganizationsOrgIdTeamsTeamIdDelete(authctx, orgid, teamid).Execute()
+	httpr, err := pco.teamclient.DefaultApi.DeleteTeam(authctx, orgid, teamid).Execute()
 	if err != nil {
 		var details string
 		if httpr != nil && httpr.StatusCode >= 400 {

--- a/go.mod
+++ b/go.mod
@@ -30,7 +30,7 @@ require (
 	github.com/mulesoft-anypoint/anypoint-client-go/secretgroup_keystore v0.1.0
 	github.com/mulesoft-anypoint/anypoint-client-go/secretgroup_tlscontext v0.1.0
 	github.com/mulesoft-anypoint/anypoint-client-go/secretgroup_truststore v0.1.0
-	github.com/mulesoft-anypoint/anypoint-client-go/team v0.3.0
+	github.com/mulesoft-anypoint/anypoint-client-go/team v1.0.1
 	github.com/mulesoft-anypoint/anypoint-client-go/team_group_mappings v0.2.1
 	github.com/mulesoft-anypoint/anypoint-client-go/team_members v0.2.0
 	github.com/mulesoft-anypoint/anypoint-client-go/team_roles v0.2.0

